### PR TITLE
Fix incorrect order summary on single product page

### LINKED ISSUE #1101: Google Pay Order Summary Doesn't Show Product Name After Selecting Shipping Address
**Describe the bug**
After clicking the Buy Now (Google Pay) button on a product page, the modal will show the product title and price. However, after choosing a Shipping Address, it recalculates and no longer shows the Product title/price separately.

**To Reproduce**
1. Enable Payment Request buttons in Stripe.
2. Open a single product page for a simple product that requires shipping.
3. Click "Buy Now" button, and see that product is listed with price:

Screenshot
Screenshot direct link: https://d.pr/i/FxWzlG

4. Choose a shipping address, and see that the product/price is no longer displayed:

Screenshot
Screenshot direct link: https://d.pr/i/ISXq0T

**Expected behavior**
Expected to still be able to see product and price broken down separately.

**Environment (please complete the following information):**
 - WordPress Version: 5.3
 - WooCommerce Version: 3.8.1
 - Stripe Plugin Version: 4.3.1

### PR DESCRIPTION
**Pull request by reykjalin**
Fixes #1101 

When using the payment request buttons (apple pay/google pay) on a product's page the order summary would initially show the product's name (itemized view). However, after changing either the shipping address or shipping method the order summary would only show the subtotal.

This fixes that behavior by passing a variable from the frontend - `is_product_page` - to the backend so we can make a decision about whether to use an itemized view (product's name in this case) or show the subtotal.

#### Changes proposed in this Pull Request:

- Passes an additional variable, `is_product_page` from the frontend when updating the shipping address or shipping method.
- `build_display_items()` now accepts a parameter, `itemized_display_items` (false by default), which can be used to force display an itemized view.
- The `is_product_page` variable is used to determine whether the customer is on the single product page, and subsequently whether to always show an itemized view or not.

You can ignore the new minified javascript file; that's there because I had to change the JS.

-------------------
- [x] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [x] Did you make changes, or create a **new .js file**? If **Gruntfile.js** exists in the repo, make sure to run `grunt`.

### CODE REVIEW

File: includes/payment-methods/class-wc-stripe-payment-request.php [hunk 1]
```diff
@@ -880,8 +889,16 @@ public function ajax_update_shipping_method() {
 
 		WC()->cart->calculate_totals();
 
+		$should_show_itemized_view = function() {
```
Comment (allendav) [inline includes/payment-methods/class-wc-stripe-payment-request.php:892]: I'm curious - why not do this imperatively?

File: includes/payment-methods/class-wc-stripe-payment-request.php [hunk 2]
```diff
@@ -787,19 +787,28 @@ public function ajax_get_shipping_options() {
 			)
 		);
 
-		$data = $this->get_shipping_options( $shipping_address );
+		$should_show_itemized_view = function() {
```
Comment (allendav) [inline includes/payment-methods/class-wc-stripe-payment-request.php:790]: Same question as below

File: includes/payment-methods/class-wc-stripe-payment-request.php (hunk 1 above)
Comment (reykjalin) [reply includes/payment-methods/class-wc-stripe-payment-request.php]: Fixed in the latest rebase https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1239/commits/a9fca6808cc3d34218b5d4975b56a876e6b8bda0 🙂 

This is a C++ habit bleeding into my PHP development here 😅 . See [this talk](https://github.com/CppCon/CppCon2017/tree/master/Presentations/10%20Core%20Guidelines%20You%20Need%20to%20Start%20Using%20Now) from cppcon 2017.

File: includes/payment-methods/class-wc-stripe-payment-request.php (hunk 2 above)
Comment (reykjalin) [reply includes/payment-methods/class-wc-stripe-payment-request.php]: See https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1239#discussion_r444550881.

**Pull request merged**